BeautifulSoupScript_edits.py

```diff
@@ -0,0 +1,213 @@
+            for pub in pubList.children:
+                #Testing to make sure pub is a <li> Tag and not an empty string
+                if not isinstance(pub, Tag):
+                    continue
+                metadata = parse_Pub_Jour(pub, journal_key)
+
+                if metadata is None:
+                    continue
+
+                downloadLink = pub.find('a', class_= "publication ss-standard ss-download btn")
+
+                if downloadLink is None:
+                    continue
+
+                metadata['url'] = downloadLink.get('href', None)
+                print(metadata)
+        '''
+        #Serializing download_dict to create a JSON file
+        with open("data_file.json", "w") as write_file:
+            json.dump(download_dict, write_file)
+        '''
+#Parsing the journal and miscelleanous publications for metadata at the bottom of webpage
+def parse_Pub_Jour(pub, journal_key):
+    pubLinks = pub.find_all('a')
+    #Link should only be the link to the publication webpage
+    detailLink = pubLinks[0].get('href', None)
+
+    detailURL = '{}{}'.format(OIC_ROOT, detailLink)
+
+    print(detailURL)
+
+    try:
+        pubElem = fetchPageHTML(detailURL)
+    except ConnectionError or HTTPError:
+        return None
+    
+    pubSoup = BeautifulSoup(pubElem.text, 'lxml')
+
+    pubContainer = pubSoup.find(class_='content-inner-left')
+
+    if pubContainer is None:
+        return None
+    #Metadata list is the last unordered list on the webpage
+    dataList = pubContainer.find_all('ul')[-1]
+
+    metadata = dataList.find_all('li')
+
+    #CHDS metadata dictionary
+    if journal_key == 2:
+        if metadata[2].text[0:4] == 'ISBN':
+            return {
+            'series': metadata[0].text,
+            'publicationInfo': metadata[1].text,
+            'isbnNumber': metadata[2].text,
+            'extent': metadata[3].text
+            }
+        else:
+            return {
+            'series': metadata[1].text,
+            'publicationInfo': metadata[2].text,
+            'isbnNumber': metadata[3].text,
+            'extent': metadata[4].text
+            }
+    #OIDA metadata dictionary
+    elif journal_key == 7:
+        return {
+        'series': metadata[0].text,
+        'publicationInfo': metadata[1].text,
+        'extent': metadata[2].text
+        }
+    #OIMP and OIS metadata dictionary
+    elif journal_key == 8  or journal_key == 10:
+        if len(metadata) == 2 or len(metadata) == 3:
+            return {
+                'publicationInfo': metadata[0],
+                'extent': metadata[1]
+            }
+        elif len(metadata) == 4: #OIP 117 has metadata in middle of page instead of at the bottom
+            return None
+        else:
+            return {
+                'series': metadata[0].text,
+                'publicationInfo': metadata[1].text,
+                'isbnNumber(hardbook)': metadata[2].text,
+                'isbnNumber(ebook)': metadata[3].text,
```

Comment:
I'd make this an array in the `isbnNumber` field for two reasons:

1. It keeps this object consistent with all of the other objects we're producing, which is important for data parsing/ingest
2. We actually treat those identifiers as essentially equivalent when it comes to processing, so it's easier down the road for us